pkg/vsphere/disk/disk_manager.go

```diff
@@ -281,7 +288,9 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 	machineSpec := types.VirtualMachineConfigSpec{}
 	machineSpec.DeviceChange = append(machineSpec.DeviceChange, changeSpec...)
 
-	m.reconfig.Lock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
```

Comment:
Pushed a new commit to introduce locking for every exported method on disk and diskmanager. DiskManager now also holds the locks during internal calls